### ISSUE TITLE
[CORE 6] `get_shared_realm()` vs `get_synchronized_realm()`

### ISSUE THREAD

**Issue by kneth** [realm/realm-object-store#848]
When porting Realm JS to core 6, I has a number of tests related to query based sync failing. The error message was `Failed to transform received changeset: Schema mismatch: Property 'permissions' in class '__Class' is an array on one side and not on the other.`. 

Moreover, I have observed that mixing file format v9 and v10 is causing issues when array of primitive types are involved.

The issues wre fixed by changing how the Realm was opened  from `new Realm()` to `Realm.open()`. One major difference between these two ways of opening a Realm is that `new Realm()` calls `get_shared_realm()` while `Realm.open()` calls `get_synchronized_realm()`.

**Comment by nhachicha** [comment]
I'm observing the same issue but for a different table 
```
09-30 12:07:29.912  9866  9894 E REALM_SYNC: Connection[1]: Session[1]: Failed to transform received changeset: Schema mismatch: Property 'members' in class '__Role' is an array on one side and not on the other.

```

**Comment by jedelbo** [comment]
The `Failed to transform received changeset` error was fixed by Sync release v4.7.1-core6.2
